pandas/_libs/tslibs/conversion.pyx

```diff
@@ -99,6 +99,11 @@ def ensure_datetime64ns(arr: ndarray, copy: bool=True):
 
     shape = (<object>arr).shape
 
+    if (<object>arr).dtype.byteorder == ">":
+        # GH#29684 we incorrectly get OutOfBoundsDatetime if we dont swap
```

Comment:
Do you know where this actually is causing the problem? Are we doing some low level byte swapping elsewhere in our code or just something getting thrown off in interfacing with numpy?

Reply:
When we do the `arr.view(np.int64)` that view is little-endian